scripts/create_burndown.py

```diff
@@ -17,211 +17,402 @@
+    return
+  
+  # Plot the water line
+  def plotWaterLine(self, x_start, x_end, y_start, y_end, color, linestyle):
+    plt.plot([x_start, x_end], [y_start, y_end], color=color, linestyle=linestyle)
+    return
+    
+  # Plot the weekend lines
+  def plotWeekendLines(self, weekend_lines, y_start, y_end, color, linestyle):
+    for weekend_line in weekend_lines:
+      plt.plot([weekend_line, weekend_line], [y_start, y_end], color=color, linestyle=linestyle)
+    return
+    
+  # Set the scale factor to display story points vs. tasks correctly in one figure
+  def setScaleFactor(self, total_tasks, max_story_points):
+    self.scalefactor = float(total_tasks) / float(max_story_points)
+    return
+
+  # Set limits for common x axis (days)
+  def setXAxisLimits(self, x_min, x_max):
+    plt.xlim([x_min, x_max])
+    return
+
+  # Set ticks and labels for the x axis
+  def setXAxisTicks(self, days):
+    x_labels = [0] + days
+    plt.xticks(days, x_labels)
+    return
+
+
+class Graph:
+  "Plot various graphs into burndown chart"
+
+  def __init__(self, data, args, config):
+    label_story_points = 'Story Points'
+    color_story_points = 'black'
+    linestyle_story_points = 'ko-'
+    linewidth_story_points = 2
+
+    label_tasks = 'Tasks'
+    color_tasks = 'green'
+    linestyle_tasks = 'go-'
+    linewidth_tasks = 2
+    
+    label_fast_lane = 'Fast Lane'
+    color_fast_lane = 'red'
+    linestyle_fast_lane = 'ro-'
+    linewidth_fast_lane = 2
+    
+    width_tasks_bar = 0.2
+    color_tasks_bar = color_tasks
+    
+    width_story_points_bar = 0.2
+    color_story_points_bar = color_story_points
+    
+    width_fast_lane_bar = 0.2
+    color_fast_lane_bar = color_fast_lane
+    
+    color_tasks_diff = color_tasks
+
+    self.setStoryPointsConfig(label_story_points, color_story_points, config.ymin, config.ymax)
+    self.plotStoryPoints(data.days, data.extra_days, data.open_story_points, data.bonus_story_points_done, linestyle_story_points, linewidth_story_points)
+    self.plotNewStoryPoints(data.total_story_points, width_story_points_bar, color_story_points_bar)
+    
+    if not args.no_tasks:
+      self.setTasksConfig(label_tasks, color_tasks, config.ymin, config.ymax, config.scalefactor)
+      self.plotTasks(data.days, data.extra_days, data.open_tasks, data.bonus_tasks_done, linestyle_tasks, linewidth_tasks)
+      self.plotNewTasks(data.total_tasks, width_tasks_bar, color_tasks_bar)
+      self.plotTasksDiff(data.total_tasks, data.open_tasks, data.bonus_tasks_day_one, data.bonus_tasks_done, data.max_story_points, data.days, config.scalefactor, color_tasks_diff)
+
+    if args.with_fast_lane:
+      self.setFastLaneConfig(label_fast_lane, color_fast_lane, config.ymin, config.ymax, config.fig)
+      self.plotFastLane(data.x_fast_lane, data.y_fast_lane, linestyle_fast_lane, linewidth_fast_lane)
+      self.plotNewFastLane(data.total_fast_lane, width_fast_lane_bar, color_fast_lane_bar)
```

Comment:
I don't think that most of this logic belongs into the `Graph` class, let alone in its constructor. The argument parsing definitely should be done by the caller of the class, not the class itself. The class and the control flow would be much simpler, if it would get less parameters. So for example instead of passing `args` the caller could do something like:

```python
plot = Plot()
...

if not args.no_tasks:
  graph = Graph("Tasks",data.tasks)
  plot.drawGraph(graph)
```